thangs_importer.py

```diff
@@ -0,0 +1,303 @@
+            modelID = str(self.model5)
+            model_title = self.modelTitle5
+        elif self.modelIndex == 6:
+            modelID = str(self.model6)
+            model_title = self.modelTitle6
+        elif self.modelIndex == 7:
+            modelID = str(self.model7)
+            model_title = self.modelTitle7
+    
+        print(self.bearer)
+        headers = {"Authorization": "Bearer "+self.bearer,}
+
+        print(headers)
+        print(self.Thangs_Config.thangs_config['url'])
+        response = requests.get(self.Thangs_Config.thangs_config['url']+"api/models/parts/"+str(modelID)+"/download-url", headers=headers)
+        print(response)
+        print(response.status_code)
+        responseData = response.json()
+        modelURL = responseData["downloadUrl"]
+        print(modelURL)
+        if modelURL is None:
+            print('Url is None')
+            return
+
+        r = requests.get(modelURL, stream=True)
+        self.uid = str(model_title)
+        temp_dir = os.path.join(Config.THANGS_MODEL_DIR, self.uid)
+        if not os.path.exists(temp_dir):
+            os.makedirs(temp_dir)
+
+        print("Temp Dir")
+        print(temp_dir)
+        # archive_path = os.path.join(temp_dir, '{}.zip'.format(uid))
+        archive_path = os.path.join(temp_dir, '{}.stl'.format(self.uid))
+        print("Arch Path")
+        print(archive_path)
+        if not os.path.exists(archive_path):
+            print("Starting Count")
+            wm = bpy.context.window_manager
+            wm.progress_begin(0, 100)
+            with open(archive_path, "wb") as f:
+                total_length = r.headers.get('content-length')
+                if total_length is None:  # no content length header
+                    f.write(r.content)
+                else:
+                    dl = 0
+                    total_length = int(total_length)
+                    for data in r.iter_content(chunk_size=4096):
+                        dl += len(data)
+                        f.write(data)
+                        done = int(100 * dl / total_length)
+                        wm.progress_update(done)
+
+            wm.progress_end()
+        else:
+            print('Model already downloaded')
+
+        # gltf_path, gltf_zip = unzip_archive(archive_path)
+        self.gltf_path = archive_path
+        self.import_model()
+        return
+        if self.gltf_path:
+            try:
+                print("Starting Import Thread")
+                # self.import_thread = threading.Thread(
+                #    target=self.import_model).start()
+                self.import_model()
+                # import_model(gltf_path, uid)
+            except Exception as e:
+                import traceback
+                print(traceback.format_exc())
+        else:
+            print("ERROR", "Download error",
+                  "Failed to download model (url might be invalid)")
+        return
+
+    def unzip_archive(self, archive_path):
```

Comment:
This method isn't used as far as I can see.  Are we going to need it for something else?

Reply:
Yes, for USDZ, I believe that George is working with it.